prow/cmd/deck/static/prow/prow.ts

```diff
@@ -393,17 +393,23 @@ function redraw(fz: FuzzySearch): void {
         return sel;
     }
 
-    function getSelectionFuzzySearch(id: string, inputId: string): string {
+    function getSelectionFuzzySearch(id: string, inputId: string): RegExp {
         const input = document.getElementById(inputId) as HTMLInputElement;
         const inputText = input.value;
-        if (inputText !== "" && opts && !opts[id + 's' as keyof RepoOptions][inputText]) {
-            return "";
+        if (inputText === "") {
+            return new RegExp('');
+        }
+        if (inputText !== "" && opts && opts[id + 's' as keyof RepoOptions][inputText]) {
+            return new RegExp(`^${inputText.replace(/[.*+?^${}()|[\]\\]/g, '\\$&')}$`);
         }
         if (inputText !== "") {
             args.push(`${id}=${encodeURIComponent(inputText)}`);
         }
-
-        return inputText;
+        const expr: string[] = [];
+        inputText.split('*').forEach((part) => {
```

Comment:
Maybe replace this declaration / `forEach` call with something like this?

```ts
const expr = inputText.split('*').map((x) => x.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'));
```

Or this, if you pull out an `escapeRegex`:

```ts
const expr = inputText.split('*').map(escapeRegex);
```